app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentInteractor.kt

```diff
@@ -118,23 +117,21 @@ class AdyenPaymentInteractor(private val adyenPaymentRepository: AdyenPaymentRep
     return inAppPurchaseInteractor.convertToFiat(amount, currency)
   }
 
-  fun mapCancellation(): Bundle {
-    return billingMessagesMapper.mapCancellation()
-  }
+  fun mapCancellation(): Bundle = billingMessagesMapper.mapCancellation()
 
-  fun removePreSelectedPaymentMethod() {
-    inAppPurchaseInteractor.removePreSelectedPaymentMethod()
-  }
+  fun removePreSelectedPaymentMethod() = inAppPurchaseInteractor.removePreSelectedPaymentMethod()
 
   fun getCompletePurchaseBundle(type: String, merchantName: String, sku: String?,
                                 purchaseUid: String?, orderReference: String?, hash: String?,
-                                scheduler: Scheduler): Single<Bundle> {
+                                scheduler: Scheduler): Single<PurchaseBundleModel> {
     val billingType = BillingSupportedType.valueOfInsensitive(type)
     return if (isManagedTransaction(billingType) && sku != null) {
       billing.getSkuPurchase(merchantName, sku, purchaseUid, scheduler, billingType)
-          .map { billingMessagesMapper.mapPurchase(it, orderReference) }
+          .map {
+            PurchaseBundleModel(billingMessagesMapper.mapPurchase(it, orderReference), it.renewal)
```

Comment:
This change should be applied to other method that are making the same mapping on the purchase flow (`LocalPaymentInteractor`, `CarrierInteractor`)